Add @CacheableTask annotation to tasks

benjaminRomano opened the pull request:
**Background**
By default gradle tasks do not cache. 

**Change**
* Add `@CacheableTask` annotation to custom tasks

**Test Plan**
* Tested `generateDebug${dbName}Interface` task properly fetched from cache

```
> Task :storage:generateDebugStorageDatabaseInterface FROM-CACHE
Custom actions are attached to task ':storage:generateDebugStorageDatabaseInterface'.
Appending implementation to build cache key: com.squareup.sqldelight.gradle.SqlDelightTask_Decorated@811fa9669948b4e5154393fe4a1f21aa
Appending additional implementation to build cache key: _BuildScript_$_run_closure2$_closure6$_closure7_5e678643d94af11f7401d007fc7e139d@dd98cb2818a7a973d14370b4ca5e675f
Appending additional implementation to build cache key: com.squareup.sqldelight.gradle.SqlDelightTask_Decorated@811fa9669948b4e5154393fe4a1f21aa
Appending additional implementation to build cache key: _BuildScript_$_run_closure2$_closure6$_closure8_1f98c42ebbcbff2ee73aa2ee42bc8837@dd98cb2818a7a973d14370b4ca5e675f
Appending input file fingerprints for 'source' to build cache key: f57eccb33f50987ceb26b5caa6f15e86
Appending output property name to build cache key: outputDirectory
Build cache key for task ':storage:generateDebugStorageDatabaseInterface' is f11976a9da37128d5d51cd46098fb235
```

AlecStrong commented:
probably worth doing what Niklas suggested before merging if thats an easy addition

benjaminRomano commented:
Added the PathSensitivity annotation to output directories and verified in our CI that the caching works correctly.

I also removed the cacheableTask annotation from the verify task. I'm not sure if it would take affect without defined a output. We don't run this in our CI so I'm unfortunately not able to test that.
